### PR TITLE
Added resource registry prp endpoint

### PR DESCRIPTION
**Pull request by TheTechArch**
Added endpoint for PRP retriavel for resource policy

### CODE REVIEW

File: src/Altinn.ResourceRegistry.Core/ResourceRegistryService.cs [hunk 1]
```diff
@@ -66,11 +66,15 @@ public async Task<List<ServiceResource>> Search(ResourceSearch resourceSearch)
...
-            Response<BlobContentInfo> response = await _policyRepository.WritePolicyAsync(filePath, fileStream);
+            Response<BlobContentInfo> response = await _policyRepository.WritePolicyAsync(serviceResources.Identifier, fileStream);
 
             return response?.GetRawResponse()?.Status == (int)HttpStatusCode.Created;
         }
+
+        /// <inheritdoc/>
+        public async Task<Stream> GetPolicy(ServiceResource serviceResource)
+        {
+              return await _policyRepository.GetPolicyAsync(serviceResource.Identifier);
```
Comment (jonkjetiloye) [inline src/Altinn.ResourceRegistry.Core/ResourceRegistryService.cs:77]: Why have the entire ServiceResource model as input if only needing the identifier?

File: src/Altinn.ResourceRegistry.Persistence/PolicyRepository.cs [hunk 2]
```diff
@@ -80,9 +85,10 @@ public async Task<Response<BlobContentInfo>> WritePolicyConditionallyAsync(strin
         }
 
         /// <inheritdoc/>
-        public async Task<string> TryAcquireBlobLease(string filepath)
+        public async Task<string> TryAcquireBlobLease(string resourceId)
         {
-            BlobClient blobClient = CreateBlobClient(filepath);
+            string filePath = $"{resourceId.AsFilePath()}/resourcepolicy.xml";
+            BlobClient blobClient = CreateBlobClient(resourceId);
```
Comment (jonkjetiloye) [inline src/Altinn.ResourceRegistry.Persistence/PolicyRepository.cs:91]: CreateBlobClient should take the filePath as input not just the resourceId.

File: src/ResourceRegistry/Controllers/ResourceController.cs [hunk 3]
```diff
@@ -83,14 +83,27 @@ public async Task<ActionResult> Post([ValidateNever] ServiceResource serviceReso
...
         [Produces("application/json")]
         [Consumes("application/json")]
-        public async Task<ActionResult> Put(ServiceResource serviceResource)
+        public async Task<ActionResult> Put(string id, ServiceResource serviceResource)
         {
+            ServiceResource currentResource = await _resourceRegistry.GetResource(id);
+
+            if (currentResource == null)
+            {
+                return BadRequest();
```
Comment (jonkjetiloye) [inline src/ResourceRegistry/Controllers/ResourceController.cs:99]: NotFound?

File: src/ResourceRegistry/Controllers/ResourceController.cs [hunk 4]
```diff
@@ -83,14 +83,27 @@ public async Task<ActionResult> Post([ValidateNever] ServiceResource serviceReso
...
+            }
+
+            if (id != serviceResource.Identifier)
+            {
+                return BadRequest();
```
Comment (jonkjetiloye) [inline src/ResourceRegistry/Controllers/ResourceController.cs:104]: Error message?

**Review by jonkjetiloye — APPROVED** [review]
Added some minor comments for consideration

File: src/Altinn.ResourceRegistry.Core/ResourceRegistryService.cs (hunk 1 above)
Comment (TheTechArch) [reply src/Altinn.ResourceRegistry.Core/ResourceRegistryService.cs:77]: Agree

**Pull request merged**